### PR TITLE
fix sort for date columns with missing values

### PR DESCRIPTION
**Pull request by nickpell**
fixes http://manage.dimagi.com/default.asp?175266

@NoahCarnahan @czue @orangejenny

### CODE REVIEW

**Comment by orangejenny** [comment]
looks reasonable once tests pass

**Comment by nickpell** [comment]
:white_check_mark:

File: corehq/apps/userreports/reports/data_source.py [hunk 1]
```diff
@@ -120,9 +122,24 @@ def get_data(self, slugs=None):
             # If a sort order is specified, sort by it.
             if self._order_by:
                 for col in reversed(self._order_by):
+                    is_descending = col[1] == DESCENDING
+                    is_date = any(
+                        configured_indicator['datatype'] == 'date'
```
Comment (czue) [inline corehq/apps/userreports/reports/data_source.py]: do you have the same problem for datetimes?

**Comment by czue** [comment]
@nickpell bump on question. seems fine to merge but if we're going to add `datetime`s in the first time we encounter this bug for them then I'd say we should just do that up front.

**Pull request merged**